packages/braid-design-system/color-mode/query-param.ts

```diff
@@ -0,0 +1,59 @@
+import { darkMode } from '../lib/css/atoms/sprinkles.css';
+
+const flag = '_bdsdm';
+// VALUES
+// 0 = light mode
+// 1 = dark mode
+// 2 = OS Preference
+declare global {
+  interface Window {
+    [flag]: '0' | '1' | '2';
+  }
+}
+
+/**
+ * Script to resolve the color mode preference from the query string, used for passing the native mobile app color mode preference to an embedded Braid web UI.
+ *
+ * Pre-minified script tag, designed to be inserted at the beginning of the document head. This ensures the color mode preference is evaluated and applied before any renderering occurs, preventing the flash of light mode.
+ *
+ * ---
+ *
+ * Example usage:
+ *
+ * &nbsp;&nbsp;&lt;html&gt;
+ *
+ * &nbsp;&nbsp;&nbsp;&nbsp;&lt;head&gt;
+ *
+ * &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&lt;!-- Place script here --&gt;
+ *
+ * ---
+ *
+ * @returns Stringified script tag to place in document head.
+ */
+export const colorModeQueryParamCheck = [
+  '<script>',
+  `((l,w)=>{try{r=/[?&]${flag}=(\\d)/;[,s]=l.search.match(r)||[];w.${flag}=s;if(s){history.replaceState(null,'',l.pathname+l.search.replace(r,'').replace(/^&/,'?')+l.hash);if(s==1||(s==2&&matchMedia('(prefers-color-scheme:dark)').matches))document.documentElement.classList.add('${darkMode}')}}catch(e){}})(location,window)`,
+  '</script>',
+].join('');
+
+/**
+ * For Braid UIs embedded within the native mobile app.
+ * Retrieves the native app color mode preference as discovered on page load from the query string.
+ *
+ * (Note: Requires the `colorModeQueryParamCheck` to have been included in the head of th document and evaluated first)
+ *
+ * ---
+ *
+ * Example usage: \``/my-url?id=123&${getColorModeQueryParam()}`\`
```

Comment:
In my VS Code the escaped backticks look off:

<img width="440" alt="with" src="https://user-images.githubusercontent.com/3297808/214482649-ac7ba52d-6c6e-497a-8bdd-e8856cdb7c58.png">

```suggestion
 * Example usage: `/my-url?id=123&${getColorModeQueryParam()}`
```

<img width="442" alt="without" src="https://user-images.githubusercontent.com/3297808/214482656-987c8742-0270-43d2-b6d5-b346fa7f2c39.png">

Reply:
Yeah this one annoyed me! Without the escaped backticks it is a template string without the quotes which is odd. I might try the code block like in your other comment though.

Reply:
```suggestion
 * Example usage: `` `/my-url?id=123&${getColorModeQueryParam()}` ``
```

Reply:
Yep, much nicer:

![Screen Shot 2023-01-25 at 4 12 59 pm](https://user-images.githubusercontent.com/912060/214485247-77b30336-0cc2-4d65-b530-cc715b5a8e7e.png)

Reply:
Even better!